File: js/get_chara_text.js
const MasterTextData_LENGTH = 5;

async function loadJson(url) {
  const res = await fetch(url);
  return await res.json();
}

async function get_Text_list() {
  const textList = [];

  for (let i = 1; i <= MasterTextData_LENGTH; i++) {
    const suffix = i === 1 ? "" : i;
    const data = await loadJson(`./masterdata/character_MasterTextData${suffix}.json`);
    const key = `CharacterText${suffix}`;
    textList.push(...data[key]);
  }

  return textList;
}

// 缓存文本数据
let textListCache = null;

async function get_Text_byId(characterId) {
  if (!textListCache) {
    textListCache = await get_Text_list();
  }
  return textListCache.filter(text => text.characterId === characterId);
}

// 示例：获取 ID 为 100101 的文本
// get_Text_byId(100101).then(console.log);

File: utils/get_chara_text.py
import json

MasterTextData_LENGTH = 5

def get_Text_list():
    text_list = []
    for i in range(1,MasterTextData_LENGTH+1):
        if i == 1:
            i = ""
        with open(f'./masterdata/character_MasterTextData{i}.json', 'r',encoding="utf-8") as file:
            data = json.load(file)[f"CharacterText{i}"]
            text_list += data
    return text_list

text_list = get_Text_list()


def get_Text_byId(characterId):
    return [text for text in text_list if text["characterId"] == characterId]
